backend/http/http.go

```diff
@@ -558,20 +623,37 @@ func (o *Object) stat(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to stat")
 	}
-	t, err := http.ParseTime(res.Header.Get("Last-Modified"))
+	var (
+		fSize int64
+		fTime time.Time
+		fType string
+	)
+	if o.fs.opt.GoIndexDriveIndex >= 0 {
+		var fInfo GoIndexFile
+		dec := json.NewDecoder(res.Body)
+		dec.Decode(&fInfo)
+		fSize = parseInt64(fInfo.Size, -1)
+		fTime, err = time.Parse(time.RFC3339, fInfo.ModifiedTime)
+		fType = fInfo.MimeType
+	} else {
+		fSize = parseInt64(res.Header.Get("Content-Length"), -1)
+		fTime, err = http.ParseTime(res.Header.Get("Last-Modified"))
+		fType = res.Header.Get("Content-Type")
+	}
 	if err != nil {
-		t = timeUnset
+		fTime = timeUnset
 	}
-	o.size = parseInt64(res.Header.Get("Content-Length"), -1)
-	o.modTime = t
-	o.contentType = res.Header.Get("Content-Type")
+	o.size = fSize
+	o.modTime = fTime
+	o.contentType = fType
 	// If NoSlash is set then check ContentType to see if it is a directory
-	if o.fs.opt.NoSlash {
+	if o.fs.opt.NoSlash || o.fs.opt.GoIndexDriveIndex >= 0 {
 		mediaType, _, err := mime.ParseMediaType(o.contentType)
 		if err != nil {
 			return errors.Wrapf(err, "failed to parse Content-Type: %q", o.contentType)
 		}
-		if mediaType == "text/html" {
+		if o.fs.opt.NoSlash && mediaType == "text/html" ||
+			o.fs.opt.GoIndexDriveIndex >= 0 && mediaType == "application/vnd.google-apps.folder" {
```

Comment:
I haven't encountered a GoIndex with online-google files (ex, a google doc) yet, so I still need to test that out, but as far as I know GoIndex doesn't support those types of files so it shouldn't be encountered / need to be handled.